feat(seo): sensible meta descriptions for agency and post pages

LinHuiqing opened the pull request:
## Problem

Meta descriptions of dynamic pages currently do not reflect the content of the pages. An example is shown below:

<img width="581" alt="Screenshot 2021-09-30 at 2 50 25 PM" src="https://user-images.githubusercontent.com/37061143/135401906-133b2242-dead-4fa9-ae95-e40d037d6919.png">

Closes #394 and #22

## Solution

Dynamically populate meta descriptions for agency-specific pages to include the agency name and post pages with the answer to the question.

**Features**:

- update meta description for post page with text of first answer
- update meta description for agency-specific page with 'Answers from `agency.longname` (`agency.shortname`)' (e.g. 'Answers from Work Allocation Singapore (WAS)')
- set meta prop `data-rh` in `index.html` to `true`
- refactor AnswerSection component to prevent repeat answer queries

**Improvements**:

- refactor PageTitle component to typescript
- replace `react-helmet` with `react-helmet-async`

## Before & After Screenshots

**BEFORE**:

On agency-specific pages:

<img width="893" alt="Screenshot 2021-09-30 at 3 18 02 PM" src="https://user-images.githubusercontent.com/37061143/135405565-b1d6ca7a-cf01-4ecb-af85-568b2763f2b6.png">

On post pages:

<img width="895" alt="Screenshot 2021-09-30 at 3 17 48 PM" src="https://user-images.githubusercontent.com/37061143/135405581-5662b668-cde5-417a-967a-07014e0a9ea6.png">

**AFTER**:

On agency-specific pages:

<img width="894" alt="Screenshot 2021-09-30 at 3 13 21 PM" src="https://user-images.githubusercontent.com/37061143/135405078-f1222c83-afe0-4607-96ef-6fd60d3a2cc3.png">

On post pages:

<img width="897" alt="Screenshot 2021-09-30 at 3 13 51 PM" src="https://user-images.githubusercontent.com/37061143/135405085-28bf0f2b-c22e-4597-b5b0-e285e8b672a2.png">

## Tests

1. Navigate to an agency-specific page and check the meta description with inspect - it should be 'Answers from `agency.longname` (`agency.shortname`)'.
2. Navigate to a post page and check the meta description with inpect - it should be the first answer to the question.
3. Navigate to the main AskGov page and check the meta description with inspect - it should be 'Answers from the Singapore Government'.

## Deploy Notes

**New dependencies**:

- `dependency`: 
  - `client/package.json`: `react-helmet-async`
  - `client/package.json`: `sanitize-html`
  - `client/package.json`: `@types/sanitize-html`

**Removed dependencies**:

- `dependency`: 
  - `client/package.json`: `react-helmet`

LinHuiqing committed:
refactor(pagetitle): migrate pagetitle to typescript

- migrate to typescript
- add description field to page title
- add package installation for @types/react-helmet

LinHuiqing committed:
feat(seo): update meta descriptions of agency and post pages

- update meta description for post page with text of first answer
- update meta description for agency-specific page with 'Answers from `agency.longname` (`agency.shortname`)'
- set meta prop data-react-helmet in index.html to true
- refactor AnswerSection component to prevent repeat answer queries

re #394

LoneRifle commented on client/public/index.html:
```diff
@@ -35,6 +35,7 @@
     <meta
       name="description"
       content="Answers from the Singapore Government"
+      data-react-helmet="true"
```
there's an outstanding issue to move to react-helmet-async. maybe we can move to that first?

LinHuiqing replied:
Okay, issue https://github.com/opengovsg/askgovsg/issues/22 is addressed in https://github.com/opengovsg/askgovsg/pull/429/commits/115944fe2300f7f43b76cd04d3a2fd426520f1fe.

LoneRifle commented on client/src/pages/Post/Post.component.jsx:
```diff
@@ -69,12 +72,22 @@ const Post = () => {
         title={`${post.title} - ${agencyShortName?.toUpperCase()} FAQ - AskGov`}
+        description={
+          answers && answers.length > 0
+            ? answers[0]?.body.replace(/<[^>]*>?/gm, '') // .replace() uses regex to remove html tags
```
can you use a sanitizer library -_-

LinHuiqing replied:
Okay :') Actually JSX already escapes all the HTML tags so there isn't any security concern I think. But to be safe 👍🏻

LinHuiqing replied:
Hm for this, I could only find a package `html-to-text`, but am facing some issues with it due to the version of webpack (refer to https://github.com/html-to-text/node-html-to-text/issues/230). Will take a break from this first to look at the previous comment.

LoneRifle replied:
Ah I misread, you're just converting HTML to text for the description. okay, deal with react-helmet-async first =)

LinHuiqing replied:
Replaced with `sanitize-html` package as suggested.

LoneRifle reviewed (changes requested):
looking good otherwise. You may want to consider populating other meta tags to improve our search ranking. eg, [keywords, Open Graph title(?)](https://gist.github.com/whitingx/3840905)

LinHuiqing committed:
perf(react-helm): migrate react-helmet to react-helmet-async

- uninstall react-helmet and @types/react-helmet
- install react-helmet-async
- update index.js and PageTitle for migration

re #22

LinHuiqing committed:
fix(post-meta): sanitize html with package instead .replace()

- .replace() might miss out dangerous tags - risk of XSS
- use sanitize-html package

LinHuiqing commented:
@LoneRifle, regarding your comments on keywords and Open Graph titles, could we move it to a new issue or separate PR? I previously read that keywords might not be effective due to the common practice of keyword stuffing, but no one can say for certain as the crawlers' algorithms are secret.

LoneRifle reviewed (approved):
lgtm, good work! you can use Facebook's Sharing Debugger to figure out what else to populate by pointing it at staging, once this PR is deployed there.
